packages/vega-util/src/mergeDeep.js

```diff
@@ -0,0 +1,33 @@
+import isArray from "./isArray";
+import isObject from "./isObject";
+import hasOwnProperty from "./hasOwnProperty";
+
+export default function mergeDeep(dest, ...src) {
+  for (const s of src) {
+    dest = mergeDeep_(dest, s);
+  }
+  return dest;
+}
+
+function mergeDeep_(dest, src) {
+  if (!isObject(src) || src === null) {
+    return dest;
+  }
+
+  for (const p in src) {
+    if (!hasOwnProperty(src, p)) {
+      continue;
+    }
+    if (src[p] === undefined) {
+      continue;
+    }
+    if (!isObject(src[p]) || isArray(src[p]) || src[p] === null) {
+      dest[p] = src[p];
+    } else if (!isObject(dest[p]) || dest[p] === null) {
+      dest[p] = mergeDeep(isArray(src[p].constructor) ? [] : {}, src[p]);
```

Comment:
1. Can't you call `mergeDeep_` (with underscore) here instead?
2. Why check the constructor instead of just `src[p]`?
3. Wouldn't `null` (rather than `[]`) work, and remove one unnecessary object instantiation.